.github/workflows/autoclose-comment.yml

```diff
@@ -0,0 +1,71 @@
+name: autoclose comment
+# Post comment on PRs when labeled with "autoclose".
+
+permissions:
+  contents: read
+  pull-requests: write
+
+on:
+  pull_request_target:
+    types:
+      - labeled
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  GH_REPO: ${{ github.repository }}
+  PULL_REQUEST_NUMBER: ${{ github.event.pull_request.number }}
+
+jobs:
+
+  post_comment:
+    name: post_comment
+    if: github.event.label.name == 'autoclose'
+    runs-on: ubuntu-latest
+
+    steps:
+
+      - name: comment on potential autoclose
+        run: |
+          gh api \
+          --method POST \
+          -H "Accept: application/vnd.github+json" \
+          -H "X-GitHub-Api-Version: 2022-11-28" \
+          /repos/$GH_REPO/issues/$PULL_REQUEST_NUMBER/comments \
+          -f "body=$BODY"
+        env:
+          BODY: |
+            This pull request has been labeled for potential autoclose.
+
+            Thank you for your contribution to scikit-learn and for the effort you've \
+            put into this PR. This pull request does not yet meet the quality and \
+            clarity needed for an effective review. Reviewing time is extremely \
+            limited, and PRs in this state are difficult to review effectively, which \
+            can slow down progress on other contributions and ongoing work. Unless \
+            this PR is improved, it will be automatically closed after two weeks.
+
+            To avoid autoclose and increase the chance of a productive review, please:
+            - Ensure your contribution aligns with our \
+            [contribution guide](https://scikit-learn.org/dev/developers/contributing.html).
+            - Include a clear motivation and concise explanation of what the PR does \
+            in the pull request description.
+            - Make sure the code runs and passes tests locally (`pytest`) and in the CI.
+            - Only submit code you understand and can maintain. We will likely request \
+            changes or explanations.
+            - Keep the changes minimal and directly relevant to the described issue or \
+            enhancement.
+
+            If you have carefully reviewed these points but are still stuck on a \
+            specific problem, please describe it in your own words. This helps \
+            maintainers see that you have designed your contribution and understand \
+            how it fits into the codebase, rather than submitting AI-generated or \
+            copied content. We especially encourage beginners to ask focused questions \
+            that show you've tried to explore the problem  and are interested to engage \
+            with the project. We may be able to provide guidance if your explanation of \
+            the problem is clear and reflects your own reasoning.
```

Comment:
I would tend to agree with Loic's suggestion of making this comment shorter. In particular this paragraph could be shortened a lot, maybe to just a one sentence bullet point in the list above.